ui/src/components/layout/EventLedger.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useState } from "react";
+import getDataStore from "../../data/DataStore";
+import { SlCard } from '@shoelace-style/shoelace/dist/react';
+import { DisplayEconomicEvent } from "../../data/models/Application/Display";
+
+export type EventLedgerProps = {};
+
+const EventLedger: React.FC<EventLedgerProps> = ({}) => {
+  const [displayEconomicEvents, setDisplayEconomicEvents] = useState<Array<DisplayEconomicEvent>>([]);
+
+  const store = getDataStore();
+
+  useEffect(()=>{
+    fetchEvents().then();
+  },[]);
+
+  const fetchEvents = async () => {
+    const events = await store.fetchAllEconomicEvents();
+    const displayEvents: Array<DisplayEconomicEvent> = [];
+    // loop through events and get names from IDs
+    for (const event of events) {
+      let displayEvent = new DisplayEconomicEvent(event);
+      await displayEvent.getNamesForId(event);
+      displayEvents.push(displayEvent);
+    }
+    setDisplayEconomicEvents(displayEvents);
+  }
+
+  const assembleCard = (econEvent: DisplayEconomicEvent) => {
```

Comment:
This is typically the kind of thing I've been placing in the `ui/logic/flows.ts` file. I've basically been building a generic sort of library that builds strings and parts of views that is agnostic of React.